### PR TITLE
Fixed Gather digits does not work without nested say/play

### PR DESCRIPTION
**Pull request by xquanluu**
#913

### CODE REVIEW

File: lib/tasks/gather.js [hunk 1]
```diff
@@ -177,8 +177,10 @@ class TaskGather extends SttTask {
     this._startVad();
 
     const startDtmfListener = () => {
+      if (this._dtmfListenerStarted) return;
```
Comment (davehorton) [inline lib/tasks/gather.js:180]: Is there a scenario where this gets called twice?

Comment (xquanluu) [reply lib/tasks/gather.js]: You are right, we don't have a case that this startDtmfListener is called twice.

Comment (davehorton) [reply lib/tasks/gather.js]: In that case can you make it an assert instead so if it ever does happen we get an error we can fix
```
assert(! this._dtmfListenerStarted);
```

File: lib/tasks/gather.js [hunk 2]
```diff
@@ -177,10 +177,8 @@ class TaskGather extends SttTask {
     this._startVad();
 
     const startDtmfListener = () => {
-      if (this._dtmfListenerStarted) return;
```
Comment (davehorton) [inline lib/tasks/gather.js]: if we are confident we are not going to call this twice, then I would still like to see an assert here

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/tasks/gather.js (hunk 2 above)
Comment (xquanluu) [reply lib/tasks/gather.js]: I see
fixed

**Review by davehorton — APPROVED** [review]

**Pull request merged**